### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by shrey4500**
Update WIP entities for entity patterns and node-entity filter.

### CODE REVIEW

File: docs/bot-builder/basic/disambiguation.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+*3 - Something else*
+
+The User can send 1, 2, 3 as a response, basis which we will detect the chosen intent by the user from the above options.
+
+**How to set up Disambiguation?** - 
+
+The text that appears on the button on the UI is controlled at a node level through a node-wise copy. It can be set through the "**Disambiguate message**" flag in the **User Says** section. 
+
+There is a **limit of 21 characters** on the User Says that can be marked as "Disambiguate message". The same has been done to ensure it can appear as a button on the Web SDK.
```
Comment (medhaanand) [inline docs/bot-builder/basic/disambiguation.md:53]: Now on the UI, Bot Builder is showing 20 characters. We would need to change this @shrey4500

File: docs/bot-builder/basic/disambiguation.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+![DMG_4](assets/DMG_4.png)
+
+**Note**: If the **Disambiguate Message Flag** is not marked on node, the IVA picks the Node name field will be displayed on the button on the Chat UI. Therefore, it’s recommended that the **Disambiguate Message Flag** is always marked on the node in the User Says section.
```
Comment (medhaanand) [inline docs/bot-builder/basic/disambiguation.md:57]: Will it still happen for the improved design or we will change the docs once that feature is out? @shrey4500

File: docs/bot-builder/basic/entities.md [hunk 3]
```diff
@@ -262,157 +264,254 @@ There are two ways in which text entities can be used.
...
-At run time, model evaluates terms based on the context in which they are used in the sentence only. If the structure of a user says that mentions the term matches the structure of an intent user example in which a mention is tagged, then the model interprets the term to be a mention of that entity type. 
+1. Vary the position of the entity in sentences in training data
 
-For example, the user input might include the utterance, **‘Play matrix reloaded on Hotstar’**. Due to the similarity of the structure of this sentence to the user example that you tagged (**Play harry potter on Hotstar**), model recognises **‘matrix reloaded’** as a **content** entity mention.
+2. Avoid spelling errors
 
-When a context based entity model is used for an entity, model does not look for exact text or pattern matches for the entity in the user input, but focuses instead on the context of the sentence in which the entity is mentioned.
+3. Ensure variety in usage of verbs. E.g if the intent is to find a course, it can be shown with verbs like look, search, find, show, etc. 
 
-The entities that are tagged in the ‘Entity Patterns’, get auto-tagged in the user says section of the node.
```
Comment (medhaanand) [inline docs/bot-builder/basic/entities.md:299]: Have we put this point anywhere?

Comment (shrey4500) [reply docs/bot-builder/basic/entities.md:299]: It's present line 306 I guess.

File: docs/bot-builder/basic/entities.md [hunk 4]
```diff
@@ -262,157 +264,254 @@ There are two ways in which text entities can be used.
...
+![EP_10](assets/EP_10.png)
 
-**Dictionary** - A set of examples for each entity . We recommend a minimum of 10 examples.
-Examples - **grocery_item**: `Guava juice`, `apples`, `ashirwaad atta`
+#### **Troubleshooting** ####
 
-##### **Patterns**
+**Entity Detection Failure - Case 1**
 
-1. Use longer sentences whenever possible. This enables model to perform better on paraphrases.
```
Comment (medhaanand) [inline docs/bot-builder/basic/entities.md:319]: Have we said this anywhere?

File: docs/bot-builder/basic/entities.md [hunk 5]
```diff
@@ -262,157 +264,254 @@ There are two ways in which text entities can be used.
...
-|GROCERY_ITEM           |
-|-----------------------|
-|Guava juice            |
-|Shuddha ashirwad atta  |
-|Pure cow ghee          |
-|Cadbury dairy milk silk|
-|Grape wine             |
+The Node-Entity Filter MAY cause a NODE to be removed from the above shortlist of NODES. But it can’t get a particular NODE added to the shortlisted NODES, if Intent Detection algorithms haven’t found such a NODE suitable. It is shown in subsequent sections below.
 
-As before, if possible, add entity values of variable lengths. For example - Guava juice (2 words), Cadbury dairy milk silk (4 words)
```
Comment (medhaanand) [inline docs/bot-builder/basic/entities.md:376]: This is a great point too....have we added this?

File: docs/bot-builder/basic/entities.md [hunk 6]
```diff
@@ -262,157 +264,254 @@ There are two ways in which text entities can be used.
...
 
-Example:
+2. **LIMITATION** - The Node-Entity Filter works only in cases where the Entity is either a **“Words or Phrases”** or a **Regex Type** and the entity should not return a dictionary, but only a specific value. For “Words or Phrases” entities, the value entered in Node-Entity Filter is **case sensitive** as well.
 
-"Show me Apple phones" and the bot is detecting "Apple" as a grocery item.
+3. Say, there are 2 entities having Node-entity Filter on a particular NODE, then the NODE will become eligible for being shortlisted only if 
+        a. The User Utterance has no entity values, or
+        b. The User Utterance has both the entity values on which filter has been applied.
 
-To fix this, Enter such phrases in the Entity Patterns of your entity without tagging anything. Such examples with nothing highlighted in them are treated as **Negative examples** and the bot learns to not extract values in such contexts.
```
Comment (medhaanand) [inline docs/bot-builder/basic/entities.md:413]: Have we added this part? Extremely important example given here.

File: docs/bot-builder/basic/entities.md [hunk 7]
```diff
@@ -262,157 +264,254 @@ There are two ways in which text entities can be used.
...
+![EP_11](assets/EP_11.png)
 
-    Examples -  
+Add variations to the Entity Patterns as shown below. They are variations of the Buy intent in 10 different forms - 
 
-    - I would like 2 packets of **cow milk**  - Entity mention in the end
-    - Help me purchase some **bananas** from that shop  - Entity mention in the middle
+![EP_12](assets/EP_12.png)
 
-4. If the entity values can vary in length in number of words, then add few variations each for different number of words
```
Comment (medhaanand) [inline docs/bot-builder/basic/entities.md:339]: IMP. point..

File: docs/bot-builder/basic/entities.md (hunk 4 above)
Comment (shrey4500) [reply docs/bot-builder/basic/entities.md:319]: Line 367

File: docs/bot-builder/basic/entities.md (hunk 7 above)
Comment (shrey4500) [reply docs/bot-builder/basic/entities.md:339]: Line 351

File: docs/bot-builder/basic/entities.md (hunk 5 above)
Comment (shrey4500) [reply docs/bot-builder/basic/entities.md:376]: Line 351

File: docs/bot-builder/basic/entities.md (hunk 6 above)
Comment (shrey4500) [reply docs/bot-builder/basic/entities.md:413]: Line 407 and 301

**Pull request merged**